src/test/scala/org/broadinstitute/hail/stats/FisherExactTestSuite.scala

```diff
@@ -0,0 +1,151 @@
+        var s = State(sc, sqlContext, vds)
+        val sampleIds = vds.sampleIds
+        val phenotypes = sampleIds.zipWithIndex.map{case (s, i) =>
+          if (i % 3 == 0)
+            (s, "ADHD")
+          else if (i % 3 == 1)
+            (s, "Control")
+          else
+            (s, "NA")
+        }
+
+        val phenotypeFile = tmpDir.createTempFile("phenotypeAnnotation",".txt")
+        writeTextFile(phenotypeFile, sc.hadoopConfiguration){case w =>
+          w.write("Sample\tPheno1\n")
+          phenotypes.foreach{case (s, p) => w.write(s"$s\t$p\n")}
+        }
+
+        s = AnnotateSamplesTable.run(s, Array("-i", phenotypeFile, "-r", "sa.pheno"))
+
+        s = AnnotateVariantsExpr.run(s, Array("-c", """va.macCase = gs.count(sa.pheno.Pheno1 == "ADHD" && g.isHet) + 2 * gs.count(sa.pheno.Pheno1 == "ADHD" && g.isHomVar)"""))
+        s = AnnotateVariantsExpr.run(s, Array("-c", """va.majCase = gs.count(sa.pheno.Pheno1 == "ADHD" && g.isHet) + 2 * gs.count(sa.pheno.Pheno1 == "ADHD" && g.isHomRef)"""))
+        s = AnnotateVariantsExpr.run(s, Array("-c", """va.macControl = gs.count(sa.pheno.Pheno1 == "Control" && g.isHet) + 2 * gs.count(sa.pheno.Pheno1 == "ADHD" && g.isHomVar)"""))
+        s = AnnotateVariantsExpr.run(s, Array("-c", """va.majControl = gs.count(sa.pheno.Pheno1 == "Control" && g.isHet) + 2 * gs.count(sa.pheno.Pheno1 == "ADHD" && g.isHomRef)"""))
+
+        s = AnnotateVariantsExpr.run(s, Array("-c", """va.fet = fet(va.macCase.toInt, va.majCase.toInt, va.macControl.toInt, va.majControl.toInt)"""))
+
+        val output = tmpDir.createTempFile("fetResults", ".tsv")
+
+        s = ExportVariants.run(s, Array("-c",
+          """VARIANT=v, macCase = va.macCase, majCase = va.majCase, macControl = va.macControl, majControl = va.majControl,
+            | pFET = va.fet.pValue, orat = va.fet.oddsRatio, lower = va.fet.ci95Lower, upper = va.fet.ci95Upper
+          """.stripMargin ,"-o", output))
+
+        val (_, q1) = s.vds.queryVA("va.macCase")
+        val (_, q2) = s.vds.queryVA("va.majCase")
+        val (_, q3) = s.vds.queryVA("va.macControl")
+        val (_, q4) = s.vds.queryVA("va.majControl")
+        val (_, q5) = s.vds.queryVA("va.fet.pValue")
+        val (_, q6) = s.vds.queryVA("va.fet.oddsRatio")
+        val (_, q7) = s.vds.queryVA("va.fet.ci95Lower")
+        val (_, q8) = s.vds.queryVA("va.fet.ci95Upper")
+
+
+        s.vds.variantsAndAnnotations.map{case (v, va) =>
+          val result = FisherExactTest(q1(va).get.asInstanceOf[Long].toInt, q2(va).get.asInstanceOf[Long].toInt,
+            q3(va).get.asInstanceOf[Long].toInt, q4(va).get.asInstanceOf[Long].toInt).result()
+          val annotationResult = Array(q5(va).asInstanceOf[Option[Double]], q6(va).asInstanceOf[Option[Double]],
+            q7(va).asInstanceOf[Option[Double]], q8(va).asInstanceOf[Option[Double]])
+
+          if (result sameElements annotationResult)
+            true
+          else
+            false
+        }.fold(true)(_ && _)
```

Comment:
Oops!  RDD doesn't have `forall?  Surprising.  Add it to RichRDD?  And `exists`.